applications/FluidDynamicsApplication/python_scripts/fluid_dynamics_analysis.py

```diff
@@ -0,0 +1,252 @@
+
+            self.InitializeSolutionStep()
+            self.SolveSingleStep()
+            self.FinalizeSolutionStep()
+
+    def InitializeSolutionStep(self):
+
+        for process in self.simulation_processes:
+            process.ExecuteInitializeSolutionStep()
+
+        if self.have_output:
+            self.output.ExecuteInitializeSolutionStep()
+
+        if self._TimeBufferIsInitialized():
+            self.solver.InitializeSolutionStep()
+
+    def SolveSingleStep(self):
+        if self._TimeBufferIsInitialized():
+            self.solver.SolveSingleStep()
+
+    def FinalizeSolutionStep(self):
+
+        if self._TimeBufferIsInitialized():
+            self.solver.FinalizeSolutionStep()
+
+        # shouldn't this go at the end of the iteration???
+        for process in self.simulation_processes:
+            process.ExecuteFinalizeSolutionStep()
+
+        if self.have_output:
+            self.output.ExecuteFinalizeSolutionStep()
+
+        if self.have_output and self.output.IsOutputStep():
+
+            for process in self.simulation_processes:
+                process.ExecuteBeforeOutputStep()
+
+            self.output.PrintOutput()
+
+            for process in self.simulation_processes:
+                process.ExecuteAfterOutputStep()
+
+        if self.save_restart:
+            self.restart_utility.SaveRestart()
+
+
+    def FinalizeAnalysis(self):
+        '''Finalize the simulation and close open files.'''
+
+        for process in self.simulation_processes:
+            process.ExecuteFinalize()
+
+        if self.have_output:
+            self.output.ExecuteFinalize()
+
+    def InitializeAnalysis(self):
+        '''Wrapper function comprising the definition of the model and the initialization of the problem.'''
+        self.SetUpModel()
+        self.SetUpAuxiliaryProcesses()
+        self.SetUpAnalysis()
+
+    def Run(self):
+        '''Wrapper function for the solution.'''
+        self.InitializeAnalysis()
+        self.RunMainTemporalLoop()
+        self.FinalizeAnalysis()
+
+if __name__ == '__main__':
+    from sys import argv
+
+    if len(argv) > 2:
+        err_msg =  'Too many input arguments!\n'
+        err_msg += 'Use this script in the following way:\n'
+        err_msg += '- With default parameter file (assumed to be called "ProjectParameters.json"):\n'
+        err_msg += '    "python fluid_dynamics_analysis.py"\n'
+        err_msg += '- With custom parameter file:\n'
+        err_msg += '    "python fluid_dynamics_analysis.py <my-parameter-file>.json"\n'
+        raise Exception(err_msg)
+
+    if len(argv) == 2: # ProjectParameters is being passed from outside
+        parameter_file_name = argv[1]
+    else: # using default name
+        parameter_file_name = "ProjectParameters.json"
+
+    with open(parameter_file_name,'r') as parameter_file:
+        parameters = Kratos.Parameters(parameter_file.read())
+
+    simulation = FluidDynamicsAnalysis(parameters)
```

Comment:
@adityaghantasala and I were wondering, why do you do this in two lines?
I did `FluidDynamicsAnalysis(parameters).Run()`, without thinking too much, but now that we saw your solutions we were curious

Reply:
I think it is just a matter of style/preference, as I diislike using "important" classes as throwaway objets. I don't really know if it has any impact (and, if it has, yours should be very sligtly faster, since python might decide to skip allocating a pointer to store the object)